service/test/_registry_test.js

```diff
@@ -1028,4 +1055,525 @@ describe('Registry', function() {
+    testErrorCallback('getModuleTwin', 'deviceId', 'moduleId');
+
+    /*Tests_SRS_NODE_IOTHUB_REGISTRY_18_001: [The `getModuleTwin` method shall throw a `ReferenceError` exception if `deviceId`, `moduleId`, or `done` is falsy. ]*/
+    testAllFalsyArgValues('getModuleTwin', 'deviceId', 0, 'deviceId', 'moduleId');
+    testAllFalsyArgValues('getModuleTwin', 'moduleId', 1, 'deviceId', 'moduleId');
+    testAllFalsyArgValues('getModuleTwin', 'done', 2, 'deviceId', 'moduleId');
+
+    /*Tests_SRS_NODE_IOTHUB_REGISTRY_18_002: [The `getModuleTwin` method shall construct an HTTP request using information supplied by the caller, as follows:
+    ```
+      GET /twins/<encodeURIComponent(deviceId)>/modules/<encodeURIComponent(moduleId)>?api-version=<version> HTTP/1.1
+      Authorization: <config.sharedAccessSignature>
+      Request-Id: <guid>
+    ```
+    ]*/
+    it('constructs a valid HTTP request', function(testCallback) {
+      var fakeHttpHelper = {
+        executeApiCall: function (method, path, httpHeaders, body, done) {
+          assert.strictEqual(method, 'GET');
+          assert.strictEqual(path, '/twins/deviceId/modules/moduleId' + endpoint.versionQueryString());
+          done(null, JSON.stringify({ deviceId: 'fakeTwin' }), { status: 200 });
+        }
+      };
+
+      var registry = new Registry(fakeConfig, fakeHttpHelper);
+      registry.getModuleTwin('deviceId', 'moduleId', testCallback);
+    });
+
+    /*Tests_SRS_NODE_IOTHUB_REGISTRY_18_003: [The `getModuleTwin` method shall call the `done` callback with a `Twin` object updated with the latest property values stored in the IoT Hub service. ]*/
+    it('calls the \'done\' callback with a \'Twin\' object', function(testCallback) {
+      var registry = new Registry({ host: 'host', sharedAccessSignature: 'sas' }, {
+        executeApiCall: function (method, path, httpHeaders, body, done) {
+          done(null, JSON.stringify({ deviceId: 'fakeTwin' }), { status: 200 });
+        }
+      });
+
+      registry.getModuleTwin('deviceId', 'moduleId', function(err, twin) {
+        assert.instanceOf(twin, Twin);
+        testCallback();
+      });
+    });
+
+  });
+
+  describe ('#updateModuleTwin', function() {
+    testErrorCallback('updateModuleTwin', 'deviceId', 'moduleId', {}, 'etag');
+
+    /*Tests_SRS_NODE_IOTHUB_REGISTRY_18_004: [The `updateModuleTwin` method shall throw a `ReferenceError` exception if `deviceId`, `moduleId`, `patch`, `etag`, or `done` is falsy. ]*/
+    testAllFalsyArgValues('updateModuleTwin', 'deviceId', 0, 'deviceId', 'moduleId', {}, 'etag');
+    testAllFalsyArgValues('updateModuleTwin', 'moduleId', 1, 'deviceId', 'moduleId', {}, 'etag');
+    testAllFalsyArgValues('updateModuleTwin', 'patch', 2, 'deviceId', 'moduleId', {}, 'etag');
+    testAllFalsyArgValues('updateModuleTwin', 'etag', 3, 'deviceId', 'moduleId', {}, 'etag');
+    testAllFalsyArgValues('updateModuleTwin', 'done', 4, 'deviceId', 'moduleId', {}, 'etag');
```

Comment:
this is starting to look really weird. I'd suggest that we keep it for now but if refactoring is needed we rethink this whole thing, maybe with an argument array or something.